include/aikido/util/detail/yaml_extension.hpp

```diff
@@ -0,0 +1,246 @@
+           << node[r].size() << ".";
+        throw runtime_error(ss.str());
+      }
+
+      for (auto c = 0u; c < cols; ++c)
+        matrix(r, c) = node[r][c].template as<_Scalar>();
+    }
+  }
+  else
+  {
+    std::stringstream ss;
+    ss << "Unknown type of matrix '" << node.Tag() << "'.";
+    throw runtime_error(ss.str());
+  }
+}
+
+//==============================================================================
+template <typename MatrixType, bool IsVectorAtCompileTime>
+struct encode_impl
+{
+  // Nothing defined. This class should be always specialized.
+
+  // The reason this exists is so we can use template specialization to switch
+  // between serializing vectors and serializing matrices based on the
+  // IsVectorAtCompileTime flag. This is a "nice to have" that lets us generate
+  // !Vector [1, 2, 3] instead of the syntactic travesty like
+  // !Matrix [[1], [2] ,[3]] when serializing an Eigen vector.
+};
+
+//==============================================================================
+// Specialization for vector type
+template <typename MatrixType>
+struct encode_impl<MatrixType, true>
+{
+  static YAML::Node encode(const MatrixType& matrix)
+  {
+    using Index = typename MatrixType::Index;
+
+    YAML::Node node;
+    node.SetTag("Vector");
+
+    for (Index i = 0; i < matrix.size(); ++i)
+      node.push_back(YAML::Node(matrix[i]));
+
+    return node;
+  }
+};
+
+//==============================================================================
+// Specialization for matrix type
+template <typename MatrixType>
+struct encode_impl<MatrixType, false>
+{
+  static YAML::Node encode(const MatrixType& matrix)
+  {
+    using Index = typename MatrixType::Index;
+
+    YAML::Node node;
+    node.SetTag("Matrix");
+
+    for (Index r = 0; r < matrix.rows(); ++r)
+    {
+      YAML::Node row(YAML::NodeType::Sequence);
+
+      for (Index c = 0; c < matrix.cols(); ++c)
+        row.push_back(matrix(r, c));
+
+      node.push_back(row);
+    }
+
+    return node;
+  }
+};
+
+} // namespace (anonymous)
+
+namespace YAML {
+
+//==============================================================================
+// Specialization for Eigen::Matrix<...>. This enables to use YAML::Node
+// with std::unordered_map as `Node(Eigen::Matrix3d::Identity())` and
+// `node.as<Eigen::Matrix3d>()`.
+template <typename _Scalar, int _Dim, int _Mode, int _Options>
+struct convert<Eigen::Matrix<_Scalar, _Dim, _Mode, _Options>>
+{
+  using MatrixType = Eigen::Matrix<_Scalar, _Dim, _Mode, _Options>;
+
+  static Node encode(const MatrixType& matrix)
+  {
+    return encode_impl<MatrixType, MatrixType::IsVectorAtCompileTime>::encode(
+        matrix);
+  }
+
+  static bool decode(
+      const YAML::Node& node,
+      Eigen::Matrix<_Scalar, _Dim, _Mode, _Options>& matrix)
+  {
+    if (node.Tag() == "Vector" || node.Tag() == "!Vector"
+        || node.Tag() == "Matrix"
+        || node.Tag() == "!Matrix")
```

Comment:
I am _pretty_ sure that we only need the set with the `!` or the one without. If I remember correctly, I have both here because the behavior changed between `yaml-cpp` 0.3 and 0.5. Aikido only supports `yaml-cpp` 0.5, so that is a moot point.

Unfortunately, I don't remember which is used in 0.5. I'd like to do a quick test and eliminate the superfluous check.